src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -90,16 +90,23 @@ private void RequestFailed(string exceptionMessage)
             WriteEvent(eventId: 3, exceptionMessage);
         }
 
-        [Event(4, Level = EventLevel.Informational)]
-        private void ConnectionEstablished(byte versionMajor, byte versionMinor)
+        [NonEvent]
+        private void ConnectionEstablished(byte versionMajor, byte versionMinor, long connectionId, string scheme, string host, int port, IPEndPoint? remoteEndPoint)
+        {
+            string? socketAddress = remoteEndPoint?.Address?.ToString();
+            ConnectionEstablished(versionMajor, versionMinor, connectionId, scheme, host, port, socketAddress);
+        }
+
+        [Event(4, Level = EventLevel.Informational, Version = 1)]
+        private void ConnectionEstablished(byte versionMajor, byte versionMinor, long connectionId, string scheme, string host, int port, string? socketAddress)
```

Comment:
Also going with the `socketAddress` name here.

Reply:
A more generic `remoteEndPoint` may be better if we do decide to allow users to customize the string that is logged here (e.g. they give us a custom stream via `ConnectCallback` that may not be backed by a socket at all).